lib/include/public/IDataInspector.hpp

```diff
@@ -118,6 +118,12 @@ namespace MAT_NS_BEGIN
         /// <param name="isGlobalContext">Whether this is a global/logmanager Context or local ILogger context</param>
         /// <param name="associatedTenant">(Optional) Tenant associated with the Context</param>
         virtual void InspectSemanticContext(const std::string& contextName, GUID_t contextValue, bool isGlobalContext, const std::string& associatedTenant) noexcept = 0;
+
+        /// <summary>
+        /// Returns unique name for current Data Inspector
+        /// </summary>
+        /// <returns>Name of Data Inspector</returns>
+        virtual const char* GetName() const noexcept { return ""; }
```

Comment:
Why are you returning a default empty string? For the interfaces, this should be set to `= 0` so the implementing class implements the method correctly.

Reply:
this was a suggestion from Matt because we need to keep PrivacyGuard building for this first PR, this will then get turned into a pure virtual function on the next update after we can fix the modules

Reply:
Ah, doh, yes. This is fine then. I'm waiting on the swap FWIW.